src/content/developers/tutorials/optimism-std-bridge-annotated-code/index.md

```diff
@@ -0,0 +1,1399 @@
+---
+title: "Optimism Standard Bridge Contract Walk-Through"
+description: How does the standard bridge for Optimism work? Why does it work this way?
+author: Ori Pomerantz
+sidebar: true
+tags: ["solidity", "optimism", "bridge", "layer 2", "l2"]
+skill: intermediate
+published: 2022-03-30
+lang: en
+---
+
+## Introduction {#introduction}
+
+[Optimism](https://www.optimism.io/) is an [Optimistic Rollup](/developers/docs/scaling/optimistic-rollups/#top).g
+Optimistic rollups can process transactions for a much lower price than the Ethereum mainnet (also known as layer 1 or L1) because transactions are only processed by a few nodes, instead of the every node on the network. 
+At the same time, the data is all writted to L1 so everything can be proved and reconstructed with all the integrity and availability guarantees of mainnet.
+
+To use L1 assets on Optimism (or any other layer 2, or L2), those assets need to be [bridged](/bridges/#prerequisites).
+One way to achieve this is for users to lock assets (ETH and [ERC-20 tokens](https://ethereum.org/en/developers/docs/standards/tokens/erc-20/) are the most common ones) on L1, and receive equivalent assets on L2. 
+Those assets can then be used, transferred, etc. on L2. 
+Eventually, whoever ends up with them might want to bridge them back to L1. 
+When doing this, the assets are burned on L2 and then released back to the user on L1.
+
+This is the way the [Optimism standard bridge](https://community.optimism.io/docs/developers/bridge/standard-bridge) works. 
+In this article we go over the source code for that bridge to see how it works and study it as an example of well written Solidity code.
+
+## Control flows {#control-flows}
+
+The bridge has two main flows:
+1. Deposit (from L1 to L2)
+1. Withdrawal (from L2 to L1)
+
+### Deposit flow {#deposit-flow}
+
+#### Layer 1
+
+1. (ERC-20 only) the depositor gives the bridge an allowance to spend the amount being deposited
+1. The depositor calls the L1 bridge (`depositERC20`, `depositERC20To`, `depositETH`, or `depositETHTo`)
```

Comment:
We should number these explicitly to avoid confusion when translating this.

```suggestion
2. The depositor calls the L1 bridge (`depositERC20`, `depositERC20To`, `depositETH`, or `depositETHTo`)
```